app/assets/javascripts/osem-schedule.js

```diff
@@ -0,0 +1,116 @@
+var url; // Should be initialize in Schedule.initialize
+var schedule_id; // Should be initialize in Schedule.initialize
+
+function showError(error){
+  // Delete other error messages before showing the new one
+  $('.unobtrusive-flash-container').empty();
+  UnobtrusiveFlash.showFlashMessage(error, {type: 'error'});
+}
+
+var Schedule = {
+  initialize: function(url_param, schedule_id_param) {
+    url = url_param;
+    schedule_id = schedule_id_param;
+  },
+  remove: function(element) {
+    var e =  $("#" + element);
+    var event_schedule_id = e.attr("event_schedule_id");
+    if(event_schedule_id != null){
+      var my_url = url + '/' + event_schedule_id;
+      var callback = function(data) {
+        console.log(data);
+        if(data.status == 'ok'){
+          e.attr("event_schedule_id", null);
+          e.appendTo($(".unscheduled-events"));
+          e.find(".schedule-event-delete-button").hide();
+        }
+        else{
```

Comment:
Why do you handle that in the success callback? I'd expect sth like

```
$.ajax({
  url: my_url,
  type: 'DELETE',
  success: success_callback,
  error: error_callback (-> eg show error),
  dataType : 'json'
+      });
```
Then you don't need the conditional in the success callback. And what I'm wondering, why does the success callback fire if it is an error?